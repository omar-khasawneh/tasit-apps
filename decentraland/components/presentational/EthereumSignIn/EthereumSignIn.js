import React from "react";
import { StyleSheet, View } from "react-native";
import PropTypes from "prop-types";
import LargeText from "@presentational/LargeText";
import WalletButton from "@presentational/WalletButton";
import Colors from "@constants/Colors";

export default function EthereumSignIn(props) {
  return (
    <View style={styles.container}>
      <LargeText>{`Cool. Let's start by hooking this app with your wallet.`}</LargeText>
      <WalletButton onConnect={props.onConnect} />
    </View>
  );
}

EthereumSignIn.propTypes = {
  onConnect: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: Colors.backgroundColor,
    flex: 1,
    justifyContent: "center",
  },
});
