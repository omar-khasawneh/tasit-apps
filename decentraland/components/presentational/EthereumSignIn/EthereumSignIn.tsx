import React from "react";
import { StyleSheet, View } from "react-native";
import LargeText from "../LargeText";
import WalletButton from "../WalletButton";
import Colors from "../../../constants/Colors";

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: Colors.backgroundColor,
    flex: 1,
    justifyContent: "center",
  },
});

const EthereumSignIn: React.FunctionComponent = () => {
  return (
    <View style={styles.container}>
      <LargeText>{`Cool. Let's start by hooking this app with your wallet.`}</LargeText>
      <WalletButton
        appName="Gnosis Safe"
        appSlug="gnosis-safe-smart-wallet"
        scheme="gnosis-safe"
      />
      <WalletButton
        appName="Tasit Wallet"
        appSlug="tasit-wallet"
        scheme="tasit-wallet"
      />
    </View>
  );
};

export default EthereumSignIn;
