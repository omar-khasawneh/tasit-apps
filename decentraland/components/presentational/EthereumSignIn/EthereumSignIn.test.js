import React from "react";
import { shallow } from "enzyme";
import EthereumSignIn from "@presentational/EthereumSignIn";

describe("EthereumSignIn", () => {
  it("renders the component", async () => {
    expect(shallow(<EthereumSignIn />)).toMatchSnapshot();
  });
});
