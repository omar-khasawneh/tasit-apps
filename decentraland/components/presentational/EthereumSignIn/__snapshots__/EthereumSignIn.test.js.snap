// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EthereumSignIn renders the component 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <LargeText>
    Cool. Let's start by hooking this app with your wallet.
  </LargeText>
  <WalletButton
    appName="Gnosis Safe"
    appSlug="gnosis-safe-smart-wallet"
    scheme="gnosis-safe"
  />
  <WalletButton
    appName="Tasit Wallet"
    appSlug="tasit-wallet"
    scheme="tasit-wallet"
  />
</Component>
`;
